urllib3/response.py

```diff
@@ -319,6 +319,8 @@ def from_httplib(ResponseCls, r, **response_kw):
         with ``original_response=r``.
         """
         headers = r.msg
+        validate_headers(headers)
```

Comment:
I don't know that that's true.

If `assert_fingerprints` fails there is a security risk, and we should explode. It's not quite so clear that failed header parsing is a security risk (it *may* be, but an argument can be made for lenient parsing).